### ISSUE TITLE
How to generate an array of a specific size?

### ISSUE THREAD

**Issue by nickknw** [jsverify/jsverify#169]
If I want an array of a random size that contains integers between 1 and 100 I can do: 

      jsc.array(js.integer(1, 100))
but in my case I want a specific number of random integers between 1 and 100. I am having trouble seeing a way to do that. It doesn't look like jsc.array accepts a max/min size parameter.

Should I instead loop within the property function and call random(1, 100) the desired number of times?
It looks like smap and/or bless are capable of creating something that does this but I don't think I quite understand them well enough to figure it out.

Any help would be much appreciated!

**Comment by phadej** [comment]
I'd happy to take accept PR adding min/max arguments to `jsc.array`

**Comment by Koleok** [comment]
Do you need the array to be of a fixed size every time?

In scenarios so far where I have needed this I have used `tuple`, but that is really only practical if you need a fixed size that is relatively small, and not if you need arrays that have say, a length of between 20 and 40.

**Comment by nickknw** [comment]
@Koleok no, I'm looking for one that could be within a range, and ideally bigger than smaller. Thanks though :)

@phadej Good to know, thanks. I took a brief look at the code and understood parts of it, so maybe sometime in the next few weeks I'll try my hand at adding that.

**Comment by danr** [comment]
I use this:

```typescript
function replicate<A>(n: number, g: jsc.Arbitrary<A>): jsc.Arbitrary<A[]> {
  const gs = [] as jsc.Arbitrary<A>[]
  for(let i=0; i<n; ++i) {
    gs.push(g)
  }
  return jsc.tuple(gs) as jsc.Arbitrary<A[]>
}
```

**Comment by leesio** [comment]
@phadej did you have a specific api/implementation in mind for this? I've been using this for some tests https://github.com/leesio/jsverify/commit/2b7c35348c0c533e8d0f303a39c98c35077bf96a
I'd be happy to tidy it up and create a PR. Alternatively, if there's a better way, I'd love to hear your thoughts.

**Comment by ben-eb** [comment]
One-liner alternative of @danr's solution:

```js
const replicate = (n, g) => jsc.tuple(Array.from(new Array(n)).map(() => g))
```

**Comment by StoneCypher** [comment]
i'm confused why that isn't `new Array(n).fill(g)`

**Comment by ben-eb** [comment]
@StoneCypher Nice! I wasn't aware of this method. 👍

**Comment by glebec** [comment]
One small addition to the short version(s) by @ben-eb / @StoneCypher. If you need to support empty array generation, you can switch to `jsc.constant`:

```js
const arrayOfLength =
    (length, arb) =>
        length === 0
            ? jsc.constant([]) // tuple disallows 0-len
            : jsc.tuple(Array(length).fill(arb))
```
